OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/README.md

```diff
@@ -190,524 +128,574 @@ In this section, you will finish developing the Office Add-in using React and Ty
-            this.ensureTable(true).then(async (tableRef:Excel.Table) => {
-                await Excel.run(async (context) => {
-                    var sheet = context.workbook.worksheets.getActiveWorksheet();
-                    // Add the new row
-                    tableRef.rows.add(null, [data]);
-                    // Autofit columns and rows if supported by API
-                    if (Office.context.requirements.isSetSupported("ExcelApi", 1.2)) {
-                        sheet.getUsedRange().format.autofitColumns();
-                        sheet.getUsedRange().format.autofitRows();
-                    }
-                    sheet.activate();
-                    return context.sync().then(() => {
-                        resolve();
-                    });
-                }).catch((err) => {
-                    reject(err);
+          }
+        }
+        ```
+
+    1. Add the following method to the `ExcelTableUtil` class.
+  
+        Notice that it calls the `ensureTable` function we just created to ensure the Excel table has been created.
+
+        ```typescript
+        // Appends a row to the table
+        addRow = async (data) => {
+          return new Promise(async (resolve, reject) => {
+            this.ensureTable(true).then(
+              async (tableRef: Excel.Table) => {
+                await Excel.run(async context => {
+                  const sheet = context.workbook.worksheets.getActiveWorksheet();
+                  // Add the new row
+                  tableRef.rows.add(null, [data]);
+                  // Autofit columns and rows if supported by API
+                  if (Office.context.requirements.isSetSupported('ExcelApi', 1.2)) {
+                    sheet.getUsedRange().format.autofitColumns();
+                    sheet.getUsedRange().format.autofitRows();
+                  }
+                  sheet.activate();
+                  return context.sync().then(() => {
+                    resolve();
+                  });
+                }).catch(err => {
+                  reject(err);
                 });
-            }, (err) => {
+              },
+              err => {
                 reject(err);
-            });
-        });
-    }
-    ````
+              }
+            );
+          });
+        }
+        ```
+
+1. Update the **App** component to leverage the methods you added to the `ExcelTableUtil` class.
+    1. Locate and open the **src/components/App.tsx** file.
+    1. Add the following `import` statement after the existing `import` statements for the the new **ExcelTableUtil** class.
+
+        ```typescript
+        import { ExcelTableUtil } from '../utils/excelTableUtil';
+        ```
+
+    1. Add the following constant after the `import` statements and update the **{{REPLACE_WITH_ALPHAVANTAGE_APIKEY}}** to use your API key.
+
+        ```typescript
+        const ALPHAVANTAGE_APIKEY: string = '{{REPLACE_WITH_ALPHAVANTAGE_APIKEY}}';
```

Comment:
might need to be updated based on @Lauragra feedback.

Reply:
See above... if this has to change, I need guidance on what service to use to collect stock quotes.